internal/controller/feature/features.go

```diff
@@ -137,6 +137,14 @@ const (
 	// Certificate resources.
 	// Github Issue: https://github.com/cert-manager/cert-manager/issues/6393
 	OtherNames featuregate.Feature = "OtherNames"
+
+	// Owner: @jsoref
+	// Alpha: v1.16
+	//
+	// UseDomainQualifiedFinalizer will enable using acme.cert-manager.io/finalizer in
+	// finalizer records instead of finalizer.acme.cert-manager.io.
+	// GitHub Issue: https://github.com/cert-manager/cert-manager/issues/7266
+	UseDomainQualifiedFinalizer featuregate.Feature = "UseDomainQualifiedFinalizer"
```

Comment:
Based on my understanding, the migration path for all this is:

1) Add support for finalizing the new finalizer in release N (but don't ever actually set it)
2) Switch the default finalizer we set/add to new objects to be the new value (but continue to recognise both) in N+x

I do like this tick-tock approach as it means we support rollback as well. `x` should probably be set to whatever our 'maximum supported rollback' is, to avoid any case where a user upgrades to behaviour in (2), creates some resources (which have the new finalizer value set) and then downgrades to a version that does *not* support the new finalizer. 

What's the purpose of the feature flag in that case then? It seems like (1) wouldn't need a feature flag, and I am not sure if it is a good idea for us to make (2) configurable either if we are trying to 'manage' this migration from old to new?

Reply:
The flag made testing _much_ easier.

It also makes migration much easier for my tiny little mind: just change the default value on the flag between releases and suddenly it does the right thing, no need to rewrite the logic.

That said, I'm not wed to using it. Anything would do.

It also means that if someone _wants_ to use this before an official release that has it on by default, they can -- and given that kubernetes screams about this stupid thing, it's kinda useful to be able to do it to reduce the number of stupid log messages.

Reply:
Would it be possible to reword this Flag/ comments to clearly state that we already support both finalizers, this flag only controls what finalizer we set when we create new resources.

Reply:
Thanks, I think I've done that.